File: VB/WindowsApplication1/CustomEditor/MyCustomBlobPopupFormViewInfo.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraEditors.Popup

Namespace WindowsApplication1
	Public Class MyCustomBlobPopupFormViewInfo
		Inherits CustomBlobPopupFormViewInfo
		Public Sub New(ByVal form As PopupBaseForm)
			MyBase.New(form)

		End Sub
		Private Function GetMyCustomButtonHeight() As Integer
			Dim form As MyPopupContainerForm = TryCast(Me.Form, MyPopupContainerForm)
			Return form.CustomButtonSize.Height + 2* form.Indent
		End Function


		Protected Overrides Function CalcSizeBarContentHeight() As Integer
			Return Math.Max(MyBase.CalcSizeBarContentHeight(), GetMyCustomButtonHeight())
		End Function


	End Class
End Namespace

File: CS/WindowsApplication1/CustomEditor/MyCustomBlobPopupFormViewInfo.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraEditors.Popup;

namespace WindowsApplication1
{
    public class MyCustomBlobPopupFormViewInfo : CustomBlobPopupFormViewInfo
    {
        public MyCustomBlobPopupFormViewInfo(PopupBaseForm form)
            : base(form)
        {
            
        }
        private int GetMyCustomButtonHeight()
        {
            MyPopupContainerForm form = Form as MyPopupContainerForm;
            return form.CustomButtonSize.Height + 2* form.Indent;
        }


        protected override int CalcSizeBarContentHeight()
        {
            return Math.Max(base.CalcSizeBarContentHeight(), GetMyCustomButtonHeight());
        }


    }
}
